draft-js-dnd-plugin/src/modifiers/onDropFile.js

```diff
@@ -0,0 +1,115 @@
+import AddBlock from './addBlock';
+import ModifyBlockData from './modifyBlockData';
+import { Entity } from 'draft-js';
+
+function readFile(file) {
+  return new Promise(resolve => {
+    const reader = new FileReader();
+
+    // This is called when finished reading
+    reader.onload = e => {
+      // Return an array with one image
+      resolve({
+        // These are attributes like size, name, type, ...
+        ...file,
+
+        // This is the files content as base64
+        src: e.target.result,
+
+        // No URL, since nothing on server
+        url: null,
+      });
+    };
+
+    reader.readAsDataURL(file);
+  });
+}
+
+function readFiles(files) {
+  return Promise.all(files.map(readFile));
+}
+
+function getBlockWhereEntityData(state, query) {
+  const map = state.getCurrentContent().get('blockMap').toArray();
+  for (let i = 0; i < map.length; i++) {
```

Comment:
we probably can use `reduce` from immutable.js instead of converting it to an array & iterating over it